diff --git a/dist/ActRunner.js b/dist/ActRunner.js
index 96d5c87e7187bdc56a7f9e4979440fcea7c83e94..68369f3913f54926ab73826262836284df72fc81 100644
--- a/dist/ActRunner.js
+++ b/dist/ActRunner.js
@@ -220,6 +220,7 @@ export class ActRunner {
                     cleanupDir(this.workingDir);
                     resolve(new ActWorkflowExecResult(code === 0 ? ActExecStatus.SUCCESS : ActExecStatus.FAILED, executionListener.getOutput(), executionListener.getJobs()));
                 });
+                process.stdout.on('data', (data) => console.log(data.toString()));
             }
             catch (err) {
                 if (err instanceof ActRunnerError) {
